src/main/java/edu/cornell/kfs/tax/batch/TaxRowClusionBuilderBase.java

```diff
@@ -0,0 +1,155 @@
+    protected final String parameterComponent;
+    protected final String transactionDocType;
+
+    protected TaxRowClusionResult resultOfPreviousCheck;
+    protected TaxRowClusionResult cumulativeResult;
+
+    protected TaxRowClusionBuilderBase(final TaxStatisticsHandler statsHandler,
+            final TaxParameterService taxParameterService, final String parameterComponent,
+            final String transactionDocType) {
+        Validate.notNull(statsHandler, "statsHandler cannot be null");
+        Validate.notNull(taxParameterService, "taxParameterService cannot be null");
+        Validate.notBlank(parameterComponent, "parameterComponent cannot be blank");
+        this.statsHandler = statsHandler;
+        this.taxParameterService = taxParameterService;
+        this.parameterComponent = parameterComponent;
+        this.transactionDocType = transactionDocType;
+        this.resultOfPreviousCheck = TaxRowClusionResult.UNKNOWN;
+        this.cumulativeResult = TaxRowClusionResult.UNKNOWN;
+    }
+
+    public TaxRowClusionResult getResultOfPreviousCheck() {
+        return resultOfPreviousCheck;
+    }
+
+    public TaxRowClusionResult getCumulativeResult() {
+        return cumulativeResult;
+    }
+
+
+
+    protected void checkAgainstExclusions(final String parameterName, final String value,
+            final TaxStatType statToUpdateIfExcluded) {
+        handleOutcomeOfExclusionCheck(
+                multiValueParameterContains(parameterName, value), statToUpdateIfExcluded);
+    }
+
+    protected void checkAgainstExcludedPrefixes(final String parameterName,
+            final String value, final TaxStatType statToUpdateIfExcluded) {
+        final Set<String> prefixes = getMultiValueParameter(parameterName);
+        handleOutcomeOfExclusionCheck(startsWithAnyIgnoreCase(value, prefixes), statToUpdateIfExcluded);
+    }
+
+    protected void checkAgainstExcludedPrefixes(final String parameterName,
+            final Stream<String> values, final TaxStatType statToUpdateIfExcluded) {
+        final Set<String> prefixes = getMultiValueParameter(parameterName);
+        handleOutcomeOfExclusionCheck(
+                values.anyMatch(value -> startsWithAnyIgnoreCase(value, prefixes)),
+                statToUpdateIfExcluded);
+    }
+
+    protected boolean startsWithAnyIgnoreCase(final String value, final Set<String> prefixes) {
+        return prefixes.stream().anyMatch(prefix -> StringUtils.startsWithIgnoreCase(value, prefix));
+    }
+
+    protected void handleOutcomeOfExclusionCheck(
+            final boolean shouldExclude, final TaxStatType statToUpdateIfExcluded) {
+        if (shouldExclude) {
+            resultOfPreviousCheck = TaxRowClusionResult.EXCLUDE;
+            cumulativeResult = TaxRowClusionResult.EXCLUDE;
+            statsHandler.increment(statToUpdateIfExcluded, transactionDocType);
+        } else {
+            resultOfPreviousCheck = TaxRowClusionResult.INCLUDE;
+        }
+    }
+
```

Comment:
There are a few extra new lines in this file that may be cleaned up

Reply:
Removing extra newlines; I was previously using them to help with code organization but I can remove them for this case.